salt/_modules/metalk8s_checks.py

```diff
@@ -16,6 +16,60 @@ def __virtual__():
     return __virtualname__
 
 
+def bootstrap_file():
+    """Check sanity of the bootstrap config file."""
+
+    networks = __pillar__.get("networks", {})
+    errors = []
+
+    # check if pod and svc CIDRs don't intersect
+    # WP and CP CIDRs
+
+    # each entry should be a list of 1 or more CIDR
+    # if not, the error is detected upon pillar generation
+    node_networks = [
+        (
+            "Workload Plane",
+            [
+                ipaddress.ip_network(cidr)
+                for cidr in networks.get("workload_plane").get("cidr")
+            ],
+        ),
+        (
+            "Control Plane",
+            [
+                ipaddress.ip_network(cidr)
+                for cidr in networks.get("control_plane").get("cidr")
+            ],
+        ),
+    ]
+
+    # These should always be defined in the pillar
+    # They are each a CIDR address
+    object_networks = [
+        ("pods", ipaddress.ip_network(networks.get("pod"))),
+        ("services", ipaddress.ip_network(networks.get("service"))),
+    ]
+
+    if len(object_networks) > 1 and object_networks[0][1].overlaps(
+        object_networks[1][1]
+    ):
+        errors.append(" - pods and services")
```

Comment:
Could be cool to have the actual CIDR in the error message (also why "Control Plane" and "Workload Plane" have capital and not "pods" and "services")